cpp/include/games/nim/Game.hpp

```diff
@@ -0,0 +1,157 @@
+  using SymmetryGroup = groups::TrivialGroup;
+  using Symmetries = core::TrivialSymmetries;
+  using Types = core::GameTypes<Constants, State, GameResults, SymmetryGroup>;
+
+  struct Rules {
+    static void init_state(State& state) {
+      state.stones_left = nim::kStartingStones;
+      state.current_player = 0;
+    }
+
+    static Types::ActionMask get_legal_moves(const StateHistory& history) {
+      const State& state = history.current();
+      Types::ActionMask mask;
+
+      for (int i = 0; i < nim::kMaxStonesToTake; ++i) {
+        mask[i] = i + 1 <= state.stones_left;
+      }
+
+      return mask;
+    }
+
+    static core::seat_index_t get_current_player(const State& state) {
+      return state.current_player;
+    }
+
+    static void apply(StateHistory& history, core::action_t action) {
+      if (action < 0 || action >= nim::kMaxStonesToTake) {
+        throw std::invalid_argument("Invalid action: " + std::to_string(action));
+      }
+
+      State& state = history.extend();
+      state.stones_left -= action + 1;
+      state.current_player = 1 - state.current_player;
+    }
+
+    static bool is_terminal(const State& state, core::seat_index_t last_player,
+                            core::action_t last_action, GameResults::Tensor& outcome) {
+      if (state.stones_left == 0) {
+        outcome.setZero();
+        outcome(last_player) = 1;
+        return true;
+      }
+      return false;
+    }
+  };
+
+  struct IO {
+    static std::string action_delimiter() { return "-"; }
+    static std::string action_to_str(core::action_t action) { return std::to_string(action); }
+    static void print_state(std::ostream&, const State&, core::action_t last_action = -1,
+                            const Types::player_name_array_t* player_names = nullptr) {
+                              throw std::runtime_error("Not implemented");
+                            }
+    static void print_mcts_results(std::ostream&, const Types::PolicyTensor& action_policy,
+                                   const Types::SearchResults&) {
+                                      throw std::runtime_error("Not implemented");
+                                   }
+    static std::string state_repr(const State& state) {
+      std::ostringstream ss;
+      ss << "[" << state.stones_left << ", " << state.current_player << "]";
+      return ss.str();
+    }
+  };
+
+  struct InputTensorizor {
+    using Tensor = eigen_util::FTensor<Eigen::Sizes<2>>;
+    using MCTSKey = State;
+    using EvalKey = State;
+
+    static MCTSKey mcts_key(const StateHistory& history) { return history.current(); }
+    template <typename Iter> static EvalKey eval_key(Iter start, Iter cur) { return *cur; }
+    template <typename Iter> static Tensor tensorize(Iter start, Iter cur) {
+
+      Tensor tensor;
+      tensor.setZero();
+      Iter state = cur;
+      tensor(0) = state->stones_left;
+      tensor(1) = state->current_player;
+      return tensor;
```

Comment:
We're not really experiment with the whole AlphaZero loop for Nim at this point, so this is fine for now. With that said, I'll make a couple comments for the future:

* I think it's likely better to tensorize `stones_left` in 0/1 with binary or unary.

* `current_player` doesn't deserve to be encoded into the tensor, since the tensors are always from the current player's point-of-view.